src/google/protobuf/compiler/csharp/csharp_source_generator_base.cc

```diff
@@ -59,7 +59,10 @@ void SourceGeneratorBase::WriteGeneratedCodeAttributes(io::Printer* printer) {
 }
 
 std::string SourceGeneratorBase::class_access_level() {
-  return IsDescriptorProto(descriptor_) ? "internal" : "public";  // public_classes is always on.
+  if(IsDescriptorProto(descriptor_)) {
+    return "internal";
+  }
+  return this->options()->internal_access ? "internal": "public";
```

Comment:
The only reason to split into `if` and `ternary` was _long line`. I'll also prefer single line.

On that note - that's the standard for line wrap at in this project - 80/100/120/none?